libslide/src/bignum/compare.rs

```diff
@@ -3,53 +3,104 @@ use std::cmp::Ordering;
 
 impl Ord for Bignum {
     fn cmp(&self, other: &Bignum) -> Ordering {
+        let (lhs, rhs) = (strip_zeros(self), strip_zeros(other));
+        // lhs and rhs have no leading or trailing (in decimal portion) zeros
         let greater: Ordering;
         let less: Ordering;
-        if self.is_neg ^ other.is_neg {
-            return other.is_neg.cmp(&self.is_neg);
-        } else if other.is_neg && self.is_neg {
+        if lhs.is_neg ^ rhs.is_neg { // need to check for negative 0
+            // use the fact that a pure zero will have an empty int and empty dec vector
+            if lhs.dec.is_empty() && rhs.dec.is_empty() && lhs.int.is_empty() && rhs.int.is_empty() {
+                return Ordering::Equal;
+            }
```

Comment:
then this is equivalent to checking `lhs_dec_term == 0 && rhs_dec_term == lhs_dec_term && ...`